src/herder/test/HerderTests.cpp

```diff
@@ -4654,31 +4656,29 @@ TEST_CASE("do not flood too many soroban transactions",
     auto otherKey = SecretKey::fromSeed(sha256("other"));
 
     SCPQuorumSet qset;
-    qset.threshold = 1;
+    qset.threshold = 2;
     qset.validators.push_back(mainKey.getPublicKey());
+    qset.validators.push_back(otherKey.getPublicKey());
 
     simulation->addNode(mainKey, qset);
     simulation->addNode(otherKey, qset);
 
-    auto updateSorobanConfig = [](Application& app) {
-        overrideSorobanNetworkConfigForTest(app);
-        modifySorobanNetworkConfig(app, [](SorobanNetworkConfig& cfg) {
-            // Update read entries to allow flooding at most 1 tx per broadcast
-            // interval.
-            cfg.mLedgerMaxReadLedgerEntries = 40;
-            cfg.mLedgerMaxReadBytes = cfg.mTxMaxReadBytes;
-        });
-    };
-
     auto app = simulation->getNode(mainKey.getPublicKey());
 
-    updateSorobanConfig(*app);
-    updateSorobanConfig(*simulation->getNode(otherKey.getPublicKey()));
-
     simulation->addPendingConnection(mainKey.getPublicKey(),
                                      otherKey.getPublicKey());
     simulation->startAllNodes();
-    simulation->crankForAtLeast(std::chrono::seconds(1), false);
+    simulation->crankForAtLeast(std::chrono::seconds(10), false);
+
+    auto cfgUpgradeSetKey = upgradeSorobanNetworkConfig(
+        [&](SorobanNetworkConfig& cfg) {
+            setSorobanNetworkConfigForTest(cfg);
+            // Update read entries to allow flooding at most 1 tx per broadcast
+            // interval.
+            cfg.mLedgerMaxReadLedgerEntries = 40;
+            cfg.mLedgerMaxReadBytes = cfg.mTxMaxReadBytes;
+        },
+        simulation);
```

Comment:
I think this test is now failing because we start flooding a little bit later (because of the upgrade), so transaction queue counts that time towards total allowed quota it can flood. You can fix this by clearing the carry-over in the transaction queue, here's the diff:
```
diff --git a/src/herder/TransactionQueue.h b/src/herder/TransactionQueue.h
index a58573a1a..4d531a68f 100644
--- a/src/herder/TransactionQueue.h
+++ b/src/herder/TransactionQueue.h
@@ -259,6 +259,14 @@ class SorobanTransactionQueue : public TransactionQueue
     }

     size_t getMaxQueueSizeOps() const override;
+#ifdef BUILD_TESTS
+    void
+    clearBroadcastCarryover()
+    {
+        mBroadcastOpCarryover.clear();
+        mBroadcastOpCarryover.resize(1, Resource::makeEmptySoroban());
+    }
+#endif

   private:
     virtual std::pair<Resource, std::optional<Resource>>
diff --git a/src/herder/test/HerderTests.cpp b/src/herder/test/HerderTests.cpp
index 03eee093e..603e57b22 100644
--- a/src/herder/test/HerderTests.cpp
+++ b/src/herder/test/HerderTests.cpp
@@ -4769,6 +4769,8 @@ TEST_CASE("do not flood too many soroban transactions",
     {
         // no broadcast right away
         REQUIRE(numBroadcast == 0);
+        tq.clearBroadcastCarryover();
+
         // wait for a bit more than a broadcast period
         // rate per period is 100 ms
         auto broadcastPeriod =
```